W-19271424-Update-DW-article-with-Gen-AI-features-CP (#499)
mulesoft/docs-code-builder: modules/ROOT/pages/int-configure-dw-expressions.adoc

If using AI-assisted transformations, evaluate these considerations:

> * To generate a DataWeave script using *Map and Transform with AI*, provide input and output samples. While optional, providing metadata helps the AI better understand the data structure and improves the accuracy of the transformation.

Reviewer: ```suggestion
* To generate a DataWeave script using *Map and Transform with AI*, provide input and output samples. While optional, providing metadata helps AI understand the data structure and improves the accuracy of the transformation.
```

Author: OK! Thanks!

* Sample data provides the context to generate functions or expressions for calculations, conditions, and other operations. You can provide Web Services Description Language (WSDL) files with sample data. See <<sample-data>> for supported data formats (MIME types) for sample data.

> * For automatic field-to-field mapping using *Map with AI*, you must provide both input and output metadata. This auto-mapping only works if there is no existing DataWeave script in the component.

Reviewer: ```suggestion
* For automatic field-to-field mapping using *Map with AI*, provide both input and output metadata. *Map with AI* works only if no DataWeave script exists in the component.
```

Author: OK! Thanks!

* AI requires valid data to create the transformation script. Write it manually if needed.
* AI chat history isn't available in *Transformation Builder*.
* Avoid using real personally identifiable information (PII) in sample data or AI inputs. Provide input and output sample data that contains only fictional or masked data.

[[map-with-ai]]
=== Map with AI

> The *Map with AI* feature automatically suggests a field-to-field mapping. This feature requires you to provide both input and output metadata for the component and only triggers if no DataWeave script is present. It performs auto-mapping based on the provided metadata.

Reviewer: ```suggestion
The *Map with AI* feature automatically suggests a field-to-field mapping. This feature requires you to provide both input and output metadata for the component and triggers only when no DataWeave script is present. It performs auto-mapping based on the provided metadata.
```

Author: OK! Thanks!
